test/TwabController.t.sol

```diff
@@ -974,13 +836,11 @@ contract TwabControllerTest is BaseTest {
 
     (newestIndex, newestObservation) = twabController.getNewestTotalSupplyObservation(mockVault);
 
-    assertEq(newestObservation.cumulativeBalance, 5771088000000e18);
```

Comment:
The cumulative balance is tested elsewhere

Each of these tests was actually testing many different things, instead of isolating each aspect of the test in separate unit tests.

Having a single giant "testFunction" test with dozens of different scenarios inside of it is poor testing practice. Instead, there should be a separate unit test for each scenario. That way, the test name can be descriptive and it become much clearer to understand what exactly the test is trying to test